protocol/xmrmaker/checks.go

```diff
@@ -171,3 +172,29 @@ func (s *swapState) checkContract(txHash ethcommon.Hash) error {
 
 	return nil
 }
+
+func (s *swapState) checkAndSetTimeouts(t0, t1 *big.Int) error {
+	s.setTimeouts(t0, t1)
+
+	if s.Backend.Env() == common.Development {
+		return nil
+	}
+
+	expectedTimeout := common.SwapTimeoutFromEnvironment(s.Backend.Env())
+	allowableTimeDiff := time.Minute * 3
+
+	if time.Now().Add(expectedTimeout).Sub(s.t0).Abs() > allowableTimeDiff {
```

Comment:
Does this work?  I thought `time.Now()` was the local time and `t0` was in UTC.  If we are going to depend on the wall-time of the client being accurate, maybe we should do a sanity check on it first? Compare it with the most recent block time? Then we can give a better error message when the client's time is off, which will definitely happen.

Reply:
they're both local time, `s.t0` is calculated with `time.Unix`: https://pkg.go.dev/time#Unix

Reply:
the alternative is to convert `time.Now()` to unix and compare to `t0`/`t1`, but these should be the same

Reply:
That makes sense. I wasn't looking at the code in my IDE when I commented. `t0` at the top is `*big.Int`, but `s.t0` is `time.Time`. Thanks.